Bencodex/Types/Dictionary.cs

```diff
@@ -11,8 +11,8 @@ namespace Bencodex.Types
 {
     public struct Dictionary :
         IValue,
-        IEquatable<IImmutableDictionary<IKey, IValue>>,
-        IImmutableDictionary<IKey, IValue>
```

Comment:
Could you explain why did you make it to no more implement `IImmutableDictionary<IKey, IValue>`?  Making all `IValue` types immutable was an intention.

Reply:
It was because return type of `Dictionary.SetItem(IKey, IValue)`can't be replaced to `Dictionary` from `IImmutableDictionary<IKey, IValue>`. Still, `Dictionary`'s Immutability is still being maintained.

Reply:
Doesn't C# allow to overload methods returning different types if they have different types of parameters?  See also: <https://dotnetfiddle.net/vXejQh>.

Reply:
No, it's allowed. I just thought about it, but I cancelled it because I was not sure that it's right way. Anyway now, it seems better to overload methods returning different types, so I will apply it.